dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+		libunwind? ( https://github.com/ClickHouse-Extras/libunwind/archive/${LIBUNWIND_COMMIT}.tar.gz -> libunwind-${LIBUNWIND_COMMIT}.tar.gz )
+		https://github.com/ClickHouse-Extras/poco/archive/${POCO_COMMIT}.tar.gz -> poco-${POCO_COMMIT}.tar.gz
+		json? (
+			!cpu_flags_x86_avx2? ( https://github.com/Tencent/rapidjson/archive/${RAPIDJSON_COMMIT}.tar.gz -> rapidjson-${RAPIDJSON_COMMIT}.tar.gz )
+			cpu_flags_x86_avx2? ( https://github.com/lemire/simdjson/archive/${SIMDJSON_COMMIT}.tar.gz -> simdjson-${SIMDJSON_COMMIT}.tar.gz )
+	)
+		static? ( https://github.com/google/re2/archive/${RE2_COMMIT}.tar.gz -> re2-${RE2_COMMIT}.tar.gz  )
+		parquet? (
+			https://github.com/apache/arrow/archive/${ARROW_COMMIT}.tar.gz -> arrow-${ARROW_COMMIT}.tar.gz
+			https://github.com/google/snappy/archive/${SNAPPY_COMMIT}.tar.gz -> snappy-${SNAPPY_COMMIT}.tar.gz
+			https://github.com/apache/thrift/archive/${THRIFT_COMMIT}.tar.gz -> thrift-${THRIFT_COMMIT}.tar.gz
+		)
+		https://github.com/ClickHouse-Extras/zlib-ng/archive/${ZLIB_NG_COMMIT}.tar.gz -> zlib-ng-${ZLIB_NG_COMMIT}.tar.gz
+	"
+	S="${WORKDIR}/${MY_PN}-${PV}-${TYPE}"
+fi
+
+RDEPEND_BASE="
+	!static? (
+		libunwind? ( !sys-libs/libunwind )
+		!dev-libs/poco
+		parquet? ( !app-arch/snappy )
+
+		libcxx? (
+			libunwind? ( sys-libs/libcxx[-libunwind] )
+		)
+		llvm? (
+			<sys-devel/llvm-8.0.0:=
+			sys-devel/clang:=[static-analyzer]
+			sys-devel/lld
+			readline? ( >=sys-devel/llvm-5.1.0:=[ncurses] )
+		)
+		mysql? (
+			|| (
+				>=dev-db/mariadb-connector-c-3.1.0
+				virtual/libmysqlclient
+			)
+		)
+		>=dev-libs/re2-0.2018.01.01:0=
+		ssl? (
+			!libressl? ( dev-libs/openssl:0= )
+			libressl? ( dev-libs/libressl:0= )
+		)
+		libedit? ( sys-libs/ncurses:0=[tinfo] )
+		readline? ( sys-libs/ncurses:0=[tinfo] )
+		dev-libs/cyrus-sasl
+	)
+	elibc_FreeBSD? ( dev-libs/libexecinfo )
+"
+
+DEPEND_BASE="${RDEPEND_BASE}
+	static? (
+		!static-libs? (
+			libunwind? ( !sys-libs/libunwind )
+			!dev-libs/poco
+			parquet? ( !app-arch/snappy )
+		)
+		libcxx? (
+			libunwind? ( sys-libs/libcxx[-libunwind] )
+		)
+		llvm? (
+			<sys-devel/llvm-8.0.0:=
+			sys-devel/clang:=[static-analyzer]
+			sys-devel/lld
+			readline? ( >=sys-devel/llvm-5.1.0:=[ncurses] )
+		)
+		mysql? (
+			|| (
+				>=dev-db/mariadb-connector-c-3.1.0[static-libs]
+				virtual/libmysqlclient[static-libs]
+			)
+		)
+		ssl? (
+			!libressl? ( dev-libs/openssl:0=[static-libs] )
+			libressl? ( dev-libs/libressl:0=[static-libs] )
+		)
+		libedit? ( sys-libs/ncurses:0=[static-libs,tinfo] )
+		readline? ( sys-libs/ncurses:0=[static-libs,tinfo] )
+		dev-libs/cyrus-sasl[static-libs]
+	)
+"
+
+BDEPEND="
+	doc? ( >=dev-python/mkdocs-1.0.1 )
+	protobuf? ( >=dev-libs/protobuf-3.6.1:= )
+	lto? (
+		|| (
+			>=sys-devel/lld-6.0.0
+			sys-devel/binutils[gold]
+		)
+	)
+	|| (
+		>=sys-devel/gcc-7.0
+		>=sys-devel/clang-6.0
+	)
+	!static? ( >=dev-util/patchelf-0.10  )
+"
+
+SLOT="0/${TYPE}"
+LLVM_MAX_SLOT=7
+
+CLICKHOUSE_DEPS=(
```

Comment:
That's not the right approach. You can't expect other devs to spend 10 minutes trying to figure out what you're doing here if they need to update the dependencies. And then another 30 minutes verifying if they don't accidentally break some fragile assumption.

Reply:
Well, there is a lot of interconnections here and it is easy to mess up anyway. Here at least I collected most of the relations in one place and while one add or edit lines so that they look similar to everything else, nothing should break without a reason.
I also tried to provide meaningful errors when something goes wrong.
If there is a "right approach" making it harder to hurt someone, I would love to change the code. At first, I was trying to just manually do all the operations via simple loops and arrays, each stage independently, but I end up with a lot of repetitions and it was too easy to miss or or two use flags.

Reply:
Please mark it as resolved or suggest an alternative